api/python/src/cellxgene_ontology_guide/ontology_parser.py

```diff
@@ -171,10 +171,8 @@ def get_term_ancestors_with_distances(self, term_id: str, include_self: bool = F
         if term_id in VALID_NON_ONTOLOGY_TERMS:
             return {}
         ontology_name = self._parse_ontology_name(term_id)
-        ancestors: Dict[str, int] = self.cxg_schema.ontology(ontology_name)[term_id]["ancestors"]
-        if include_self:
-            ancestors[term_id] = 0
-        return ancestors
```

Comment:
this was the offending code in the example in the issue, it is called with include_self=True in get_lowest_common_ancestors

Reply:
thanks @nayib-jose-gloria . I think i tracked it here too, but couldn't quite parse out what was happening that made it stateful. Thanks for resolving this!